pkg/model/job.go

```diff
@@ -220,11 +221,17 @@ type PublisherSpec struct {
 	Params map[string]interface{} `json:"Params,omitempty"`
 }
 
+type EngineSpec struct {
+	Type   string                 `json:"Type,omitempty"`
+	Params map[string]interface{} `json:"Params,omitempty"`
+}
+
 // Spec is a complete specification of a job that can be run on some
 // execution provider.
 type Spec struct {
-	// e.g. docker or language
-	Engine Engine `json:"Engine,omitempty"`
+	// deprecated: user EngineSpec instead.
+	EngineDeprecated Engine `json:"Engine,omitempty"`
+	EngineSpec       EngineSpec
```

Comment:
Is there any point deprecating this if the Docker and Wasm fields are completely gone? As that change will already have broken every previous job.

Reply:
I will remove these fields from this version of the job spec and also preserve the previous job spec schema to maintain backwards compatibility as described in https://github.com/bacalhau-project/bacalhau/discussions/2605. Doing that now in the following commits.